icu4c/source/test/intltest/localebuildertest.cpp

```diff
@@ -190,6 +197,9 @@ void LocaleBuilderTest::TestLocaleBuilder() {
         status = U_ZERO_ERROR;
         bld.clear();
         while (true) {
+            status = U_ZERO_ERROR;
+            UErrorCode copyStatus = U_ZERO_ERROR;
+            bld.copyErrorTo(copyStatus);
```

Comment:
You copy the error before you start mutating the LocaleBuilder.  Wouldn't you want to put this after you call all the various builder methods?

I would also like to see a concrete test where the error code is not just U_ZERO_ERROR.  Give it bad input on purpose, and check that the error code is as expected.  Maybe you already have that, but it's not obvious from looking at the test.